Fixed issues to run integration test

zhanggbj opened the pull request:
This patch fixed several issue to run integration test with real server.
1. Split bmp client url to use softlayer-go http client
2. Always print message to console using fmt without specifying  --verbose if using cmd.ui
3. Add session timeout to get the response
4. Expect RunBmpLogin() return correct response

[#118921515](https://www.pivotaltracker.com/story/show/118921515)

Commit:
Fixed issues to run integration test

[#118921515] (https://www.pivotaltracker.com/story/show/118921515)

maximilien commented on clients/bmp_client.go:
```diff
@@ -23,8 +24,19 @@ type bmpClient struct {
 	var httpClient softlayer.HttpClient
+
+	useHttps := false
+	if url != "" {
```
Cool, but move this to a utility public method in `common` named like `IsHttpsUrl(url string) bool` or similarly named

zhanggbj replied:
Actually I'm intended to remove the "http://" in the url here, so the http client can concat the right url to call the server, not just to check whether it is using https or not. So I think it's better to keep it here:)

zhanggbj replied:
Added it to a private method.

maximilien replied:
Cool 👍

maximilien commented on cmds/bmp/login_command.go:
```diff
@@ -86,5 +87,6 @@ func (cmd loginCommand) Execute(args []string) (int, error) {
 		return 1, err
 	}
 
+	fmt.Println("Login Successfully!")
```
I'd say this message should be: `"Login successful"` since `"Login Successfully"` is not correct English

zhanggbj replied:
Done

maximilien commented on cmds/bmp/status_command.go:
```diff
@@ -56,11 +57,8 @@ func (cmd statusCommand) Execute(args []string) (int, error) {
-	cmd.ui.Printf(" name: %s", info.Data.Name)
-	cmd.ui.Printf(" version: %s", info.Data.Version)
-	cmd.ui.Println()
+	fmt.Println(" name:    ", info.Data.Name)
```
Let's discuss this change. The cmd.ui is meant to encapsulate all output and I feel if you are doing this due to debugging then we may need to fix the `Ui` interface

maximilien commented on integration/test_helpers/helpers.go:
```diff
@@ -47,7 +43,7 @@ func RunCommand(cmd string, args ...string) *Session {
 	session, err := Start(command, GinkgoWriter, GinkgoWriter)
 	Expect(err).NotTo(HaveOccurred())
-	session.Wait()
+	session.Wait(10 * time.Second)
```
Let's keep all waits in constant at top of file since I want these to be as small as possible (i.e., need tests to be as fast as possible) and making sure we can change these easily to try smaller values

zhanggbj replied:
Done. Also changed timeout to 3 seconds and this works for current integration test.

Commit:
Addressed PR comments

Addressed comments in PR #38

[#118921515] (https://www.pivotaltracker.com/story/show/118921515)

maximilien commented on clients/bmp_client.go:
```diff
@@ -287,3 +294,15 @@ func (bc *bmpClient) CreateBaremetal(createBaremetalInfo CreateBaremetalInfo, dr
+// Private methods
+
+func analyseURL(url string) (bool, string) {
+	s := strings.Split(url, "://")
```
use descriptive names for variables, `s` and `u` are terrible names.

zhanggbj replied:
done

maximilien commented on cmds/bmp/bms_command.go:
```diff
@@ -115,7 +114,7 @@ func (cmd bmsCommand) Execute(args []string) (int, error) {
 		table.Append(value)
 	}
-	cmd.ui.PrintTable(table)
+	table.Render()
```
I don't really like this. Reason is that now the table has to be created with `os.Stdout` and I want to have all output controlled by `cmd.Ui` interface. One thing you could do is to have the `Ui` interface return a new `TableWriter` and use there the `os.Stdout`. So like 98 becomes:

```
table := cmd.ui.NewTableWriter()
```

zhanggbj replied:
done

maximilien commented on cmds/bmp/stemcells_command.go:
```diff
@@ -76,8 +75,8 @@ func (cmd stemcellsCommand) Execute(args []string) (int, error) {
-	cmd.ui.Println(table)
-	cmd.ui.Println("Stemcells total:", length)
+	cmd.ui.PrintTable(table)
+	cmd.ui.PrintlnInfo("Stemcells total: ", length)
```
On the other hand, this is good 👍

Commit:
Minor changes

1.Created cmd.ui.NewTableWriter
2.Minor changes

[##118921515] (https://www.pivotaltracker.com/story/show/118921515)

Commit:
Merge remote-tracking branch 'upstream/master' into fixissue

Commit:
Updated travis issue

maximilien commented on common/default_printer.go:
```diff
@@ -31,9 +34,20 @@ func (p defaultPrinter) Println(args ...interface{}) (int, error) {
+}
+
+func (p defaultPrinter) NewTableWriter() *tablewriter.Table {
+	table := tablewriter.NewWriter(os.Stdout)
```
Just make this into one line. I'll accept but change this next chance you get.
